shared/src/types/key/mod.rs

```diff
@@ -356,6 +357,26 @@ impl<PK: PublicKey> From<&PK> for PublicKeyHash {
     }
 }
 
+/// Convert validator's consensus key into address raw hash that is compatible
+/// with Tendermint
+pub fn tm_consensus_key_raw_hash(pk: &common::PublicKey) -> String {
+    match pk {
+        common::PublicKey::Ed25519(pk) => {
+            let pkh = PublicKeyHash::from(pk);
+            pkh.0
+        }
+        common::PublicKey::Secp256k1(pk) => {
+            let pkh = PublicKeyHash::from(pk);
+            pkh.0
+        }
+    }
+}
+
+/// Convert Tendermint validator's raw hash bytes to Namada raw hash string
+pub fn tm_raw_hash_to_string(raw_hash: impl AsRef<[u8]>) -> String {
+    HEXUPPER.encode(raw_hash.as_ref())
+}
```

Comment:
I'm not sure about these helpers here, could they live with the Tendermint related code that will use them? (they aren't used yet in this PR). These might belong in a separate PR

Reply:
Those got grabbed when I synced this file with main. There was a bug when looking up tendermint addresses (@sug0  should remember. Something about not being able to slash if someones address wasn't valid utf-8). I can remove these as they aren't directly relevant to this PR, but they will reappear if we ever merge in main.

Reply:
Oh right, I thought these were being added new for some further PR, if they are already merged in `main` then I guess it's immaterial. I'm guessing the bug was re. https://github.com/anoma/namada/blob/a669ca573f1d3f61b87cdcb8eba7cb9a6f872a6a/shared/src/types/vote_extensions/ethereum_events.rs#L24-L27

Reply:
I've opened https://github.com/anoma/namada/issues/635. I don't fully recall the issue either but I think it was that a validator should be able to look at it's own storage to get a validator's address, rather than trust an address provided in vote extensions.

Reply:
we were trying to parse utf8 data from raw hashes, which are of course completely arbitrary, and there was a piece of code that effectively skipped slashing if you failed to parse a hash as utf8 data.